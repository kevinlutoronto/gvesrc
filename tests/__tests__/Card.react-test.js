/* eslint-disable */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

let Card;

describe('Card Component snapshots', () => {
  beforeEach(() => {
    jest.mock('react-dom', () => {
      return {
        findDOMNode: ref => {
          return { offsetHeight: 80 };
        },
      };
    });
    Card = require('../../components/Card/Card').default;
  });

  test('render', () => {
    const component = renderer.create(<Card>YOLO</Card>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('expanded', () => {
    const expandedComponent = <div>expanded</div>;
    const component = renderer.create(<Card expandedComponent={expandedComponent}>YOLO</Card>);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.isExpanded = true;

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Card Component unit', () => {
  beforeEach(() => {
    Card = require('../../components/Card/Card').default;
  });

  test('children', () => {
    const component = shallow(<Card>YOLO</Card>);
    expect(component.contains('YOLO')).toBe(true);
  });

  test('min width', () => {
    const component = shallow(<Card minWidth={100}>YOLO</Card>);
    expect(component.prop('style').minWidth).toBe(100);
  });

  test('no min width', () => {
    const component = shallow(<Card>YOLO</Card>);
    expect(component.prop('style').minWidth).toBe(undefined);
  });
  test('expanded', () => {
    const expandedComponent = <div>expanded</div>;
    const component = shallow(
      <Card isExpanded expandedComponent={expandedComponent}>
        YOLO
      </Card>
    );

    expect(component.is('.isExpanded')).toBe(true);
    expect(component.find('.expanded')).toHaveLength(1);
  });
  test('actions', () => {
    const actions = [
      <div key="action1" className="action" />,
      <div key="action2" className="action" />,
    ];
    const component = shallow(<Card actions={actions}>YOLO</Card>);
    expect(component.find('.action')).toHaveLength(2);
  });

  test('no red border color', () => {
    const component = shallow(<Card>Test Card</Card>);
    expect(component.is('.borderRed')).toBe(false);
  });

  test('red border color', () => {
    const component = shallow(<Card borderColor="red">Test Red Card</Card>);
    expect(component.is('.borderRed')).toBe(true);
  });
});
