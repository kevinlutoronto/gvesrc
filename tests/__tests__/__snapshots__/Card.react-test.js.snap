// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component snapshots expanded 1`] = `
<div
  className="card"
  onClick={undefined}
  style={
    Object {
      "minWidth": undefined,
    }
  }
>
  <div
    className="inner"
  >
    <div
      className="default"
      style={
        Object {
          "maxHeight": 80,
        }
      }
    >
      YOLO
    </div>
    <div
      className="expanded"
      style={
        Object {
          "maxHeight": 0,
        }
      }
    >
      <div>
        expanded
      </div>
    </div>
  </div>
</div>
`;

exports[`Card Component snapshots expanded 2`] = `
<div
  className="card"
  onClick={undefined}
  style={
    Object {
      "minWidth": undefined,
    }
  }
>
  <div
    className="inner"
  >
    <div
      className="default"
      style={
        Object {
          "maxHeight": 80,
        }
      }
    >
      YOLO
    </div>
    <div
      className="expanded"
      style={
        Object {
          "maxHeight": 0,
        }
      }
    >
      <div>
        expanded
      </div>
    </div>
  </div>
</div>
`;

exports[`Card Component snapshots render 1`] = `
<div
  className="card"
  onClick={undefined}
  style={
    Object {
      "minWidth": undefined,
    }
  }
>
  <div
    className="inner"
  >
    <div
      className="default"
      style={
        Object {
          "maxHeight": undefined,
        }
      }
    >
      YOLO
    </div>
  </div>
</div>
`;
